pkg/docker/docker.go

```diff
@@ -864,54 +1073,60 @@ func (d *stiDocker) GetImageID(name string) (string, error) {
 // CommitContainer commits a container to an image with a specific tag.
 // The new image ID is returned
 func (d *stiDocker) CommitContainer(opts CommitContainerOptions) (string, error) {
-	repository, tag := docker.ParseRepositoryTag(opts.Repository)
-	dockerOpts := docker.CommitContainerOptions{
-		Container:  opts.ContainerID,
-		Repository: repository,
-		Tag:        tag,
+	ctx, cancel := getDefaultContext(DefaultDockerTimeout)
+	defer cancel()
+	dockerOpts := dockertypes.ContainerCommitOptions{
+		Reference: opts.Repository,
 	}
 	if opts.Command != nil || opts.Entrypoint != nil {
-		config := docker.Config{
+		config := dockercontainer.Config{
 			Cmd:        opts.Command,
 			Entrypoint: opts.Entrypoint,
 			Env:        opts.Env,
 			Labels:     opts.Labels,
 			User:       opts.User,
 		}
-		dockerOpts.Run = &config
+		dockerOpts.Config = &config
 		glog.V(2).Infof("Committing container with dockerOpts: %+v, config: %+v", dockerOpts, config)
 	}
 
-	image, err := d.client.CommitContainer(dockerOpts)
-	if err == nil && image != nil {
-		return image.ID, nil
+	resp, err := d.client.ContainerCommit(ctx, opts.ContainerID, dockerOpts)
+	if err == nil {
+		return resp.ID, nil
 	}
 	return "", err
 }
 
 // RemoveImage removes the image with specified ID
 func (d *stiDocker) RemoveImage(imageID string) error {
-	return d.client.RemoveImage(imageID)
+	_, err := d.kubeDockerClient.RemoveImage(imageID, dockertypes.ImageRemoveOptions{})
+	return err
 }
 
 // BuildImage builds the image according to specified options
 func (d *stiDocker) BuildImage(opts BuildImageOptions) error {
-	dockerOpts := docker.BuildImageOptions{
-		Name:                opts.Name,
-		NoCache:             true,
-		SuppressOutput:      false,
-		RmTmpContainer:      true,
-		ForceRmTmpContainer: true,
```

Comment:
actually I found them when trying to answer your question ...the field name did not include `TmpContainer` and that threw me off.

Adding as part of next push.